src/apps/allbridge/shortcuts.ts

```diff
@@ -0,0 +1,92 @@
+import { Address, encodeFunctionData, erc20Abi, parseUnits } from 'viem'
+import { z } from 'zod'
+import { getClient } from '../../runtime/client'
+import { ZodAddressLowerCased } from '../../types/address'
+import { NetworkId } from '../../types/networkId'
+import {
+  createShortcut,
+  ShortcutsHook,
+  Transaction,
+} from '../../types/shortcuts'
+import { poolAbi } from './abis/pool'
+
+// Hardcoding for now
+const GAS = 1_000_000n
+
+const hook: ShortcutsHook = {
+  async getShortcutDefinitions(networkId: NetworkId) {
+    return [
+      createShortcut({
+        id: 'deposit',
+        name: 'Deposit',
+        description: 'Lend your assets to earn interest',
+        networkIds: [networkId],
+        // category: 'deposit',
+        triggerInputShape: {
+          token: z.object({
+            // TODO: consider requiring only tokenId and (decimal) amount
+            // Right now it would mean more changes in hooks
+            address: ZodAddressLowerCased,
+            decimals: z.coerce.number(),
+            amount: z.string(), // in decimal string
+          }),
+          positionAddress: ZodAddressLowerCased,
+        },
+
+        async onTrigger({ networkId, address, token, positionAddress }) {
+          const walletAddress = address as Address
+          const transactions: Transaction[] = []
+
+          // amount in smallest unit
+          const amountToSupply = parseUnits(token.amount, token.decimals)
```

Comment:
Does this make the user deposit their full amount of the deposit token?

Reply:
In the CLI the amount deposited is specified by `--token.amount` flag. I am unsure of how this flag is used by Valora within the dapp shortcuts page.

Reply:
Yes there's no UI yet, earn shortcuts will be the first to pass those arguments.